include/ofi_mr.h

```diff
@@ -327,6 +327,30 @@ bool ofi_mr_cache_flush(struct ofi_mr_cache *cache, bool flush_lru);
 
 int ofi_mr_cache_search(struct ofi_mr_cache *cache, const struct fi_mr_attr *attr,
 			struct ofi_mr_entry **entry);
+
+enum ofi_mr_cache_status{
+	OFI_MR_CACHE_STATUS_NORMAL                = 0,
+	OFI_MR_CACHE_STATUS_EXCEEDS_COUNT         = (1 << 0),
+	OFI_MR_CACHE_STATUS_EXCEEDS_SIZE          = (1 << 1),
+	OFI_MR_CACHE_STATUS_FLUSH_ENTRIES_WAITING = (1 << 2),
+};
+
+static inline unsigned int ofi_mr_cache_status_evaluate(struct ofi_mr_cache * cache)
+{
+	/* mm_lock mutex must be held during this call */
```

Comment:
Unrelated note, we may want to abstract the mutex so that we can check that the lock is held.  I added something similar to fastlock and the ep lock, which turned out useful to find bugs, but also document locking requirements explicitly in the code, versus using a comment.

Reply:
Can you point at a specific section in the code where this has been done today? I've seen comments sprinkled all through the code base with similar notes. I have no problem asking this of Mike as long as we are trying to make things better going forward.

Reply:
Yeah, there were many places were there are comments like this.  To help,  I added a macro fastlock_held() (see ofi_lock.h):
```
static inline int fastlock_held(fastlock_t *lock)
{
	return lock->in_use;
}
```
I replaced comments in the code with direct checks, such as this:
```
struct rxm_tx_buf *rxm_get_tx_buf(struct rxm_ep *ep)
{
	struct rxm_tx_buf *buf;

	assert(fastlock_held(&ep->util_ep.lock));
```
We don't have anything similar for pthread_mutex because we're not abtracting it.  That would be a fair amount of work beyond this PR.